solr/core/src/java/org/apache/solr/filestore/DistribPackageStore.java

```diff
@@ -94,8 +94,18 @@ private static Path _getRealPath(String path, Path solrHome) {
     if (!path.isEmpty() && path.charAt(0) != File.separatorChar) {
       path = File.separator + path;
     }
-    // Use concat because path might start with a slash and be incorrectly interpreted as absolute
-    return solrHome.resolve(PackageStoreAPI.PACKAGESTORE_DIRECTORY + path);
+    if (path.startsWith("\\\\")) { // Windows absolute UNC
+      throw new SolrException(BAD_REQUEST, "Illegal path " + path);
+    }
```

Comment:
There was no sanitizing of the path value here, which could lead to path traversal or Windows UNC attacks

Reply:
If we have this same check somewhere else (as I think you did in other work in the past months), it'd be nice to have this in one utility method somewhere.

Reply:
You have a good memory :) I did extract the UNC check into a separate method in `SolrPaths#assertNotUnc`.